collector/overhead_alert_metrics.go

```diff
@@ -24,7 +25,7 @@ func SetupOverheadController(mgr ctrl.Manager) error {
 		eventRecorder: mgr.GetEventRecorderFor("MetricExporterExecutionOverhead"),
 		collector:     NewOverheadCollector(),
 	}
-	return ctrl.NewControllerManagedBy(mgr).For(&v1.PipelineRun{}).WithEventFilter(&taskRunGapEventFilter{}).Complete(reconciler)
+	return ctrl.NewControllerManagedBy(mgr).For(&v1.PipelineRun{}).WithEventFilter(&overheadGapEventFilter{client: reconciler.client}).Complete(reconciler)
```

Comment:
I decided to not filter our more granular metric, which is no longer used for the alerts, to allow for points of comparison.